### PR TITLE
Scheduling docs update

### PR DESCRIPTION
**Pull request by mrocklin**
Going through this commit-by-commit might make sense

### CODE REVIEW

**Comment by mrocklin** [comment]
@jcrist this could use your OK on the diagnostics pages

**Comment by mrocklin** [comment]
Merging in 24 hours otherwise

File: dask/diagnostics/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from __future__ import absolute_import, division, print_function
 
+from ..callbacks import Callback
```
Comment (jcrist) [inline dask/diagnostics/__init__.py]: Why this change?

File: docs/source/diagnostics-single.rst [hunk 2]
```diff
@@ -1,54 +1,29 @@
-Diagnostics
-===========
+Diagnostics (single)
```
Comment (jcrist) [inline docs/source/diagnostics-single.rst]: Perhaps `Diagnostics (local)` or `Diagnostics (single machine)`? Current title sounds off.

File: docs/source/scheduling.rst [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+For example, when using IPython or Jupyter notebooks the ``%debug``, ``%pdb``, or ``%prun`` magics
+will not work well when using the parallel dask schedulers;
+they were not designed to be used in a parallel computing context.
+However if you run into an exception and want to step into the debugger
+you may wish to rerun your computation under the single-threaded scheduler,
+where these tools will function properly.
+
+
+Dask.Distributed (local)
```
Comment (jcrist) [inline docs/source/scheduling.rst]: Using a module path with title case seems wrong to me. Either `dask.distributed (local)` or `Dask Distributed (local)` sound better. Same for below.

File: docs/source/graphviz.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
+Visualize task graphs
```
Comment (jcrist) [inline docs/source/graphviz.rst]: Since this basically replaces the `inspect.rst` page, should probably delete that page as well.

File: docs/source/graphviz.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+.. autosummary::
+   visualize
+
+Before executing your computation you might consider visualizing the underlying task graph.
+By looking at the inter-connectedness of tasks
+you can learn more about potential bottlenecks
+where parallelism may not be possile,
+or areas where many tasks depend on each other,
+which may cause a great deal of communication.
```
Comment (jcrist) [inline docs/source/graphviz.rst]: Not-important-nit: line lengths/wraps vary a lot in this PR. When I write docs in vim I often run `gqap` per paragraph to nicely wrap to the 80 char line limit. Other editors probably can do this as well. I (personally) find it easier to read the rst docs when consistent formatting is applied to the source. Not a big deal though, no need to change anything.

File: docs/source/diagnostics-single.rst [hunk 6]
```diff
@@ -148,7 +129,14 @@ the number of bytes in the scheduler cache:
...
 
 Example
-^^^^^^^
+-------
+
+.. autosummary::
+   Profiler
+   ResourceProfiler
+   CacheProfiler
+   visualize
```
Comment (jcrist) [inline docs/source/diagnostics-single.rst]: These are already autosummaried individually above, here, and then again at the bottom. Personally I'd only do the one at the bottom, but at a minimum I'd delete this autosummary here.

This also feels like a weird spot for the summary, as its right after the title `Example`, but before the example.

File: dask/diagnostics/__init__.py (hunk 1 above)
Comment (mrocklin) [reply dask/diagnostics/__init__.py]: It seems sensible to me to have `Callback` available in the `dask.diagnostics` namespace.  Happy to revert though.

File: docs/source/graphviz.rst (hunk 5 above)
Comment (mrocklin) [reply docs/source/graphviz.rst]: Ah, I started doing this in my personal writing and have been enjoying it.  Happy to revert though, it's not standard in this codebase.  http://rhodesmill.org/brandon/2012/one-sentence-per-line/

File: docs/source/diagnostics-single.rst (hunk 6 above)
Comment (mrocklin) [reply docs/source/diagnostics-single.rst]: I tend to do a little bit of auto-summary in each section with the functions relevant to that section.  I think that this is done particularly well in http://dask.pydata.org/en/latest/futures.html

File: docs/source/scheduling.rst (hunk 3 above)
Comment (mrocklin) [reply docs/source/scheduling.rst]: Agreed.  Thanks for the suggestion.  I think I've hit all of the various links.

File: docs/source/diagnostics-single.rst (hunk 2 above)
Comment (mrocklin) [reply docs/source/diagnostics-single.rst]: Agreed. Thanks for the suggestion. I think I've hit all of the various links.

File: docs/source/scheduling.rst (hunk 3 above)
Comment (mrocklin) [reply docs/source/scheduling.rst]: Done

File: docs/source/graphviz.rst (hunk 5 above)
Comment (jcrist) [reply docs/source/graphviz.rst]: Ah, interesting. No strong thoughts either way. Up to you.

File: dask/diagnostics/__init__.py (hunk 1 above)
Comment (jcrist) [reply dask/diagnostics/__init__.py]: Fine by me.

File: docs/source/graphviz.rst (hunk 4 above)
Comment (jcrist) [reply docs/source/graphviz.rst]: I think this is the only thing remaining ^^

**Comment by mrocklin** [comment]
Thanks for the review @jcrist !

File: dask/base.py [hunk 7]
```diff
@@ -93,6 +93,18 @@ def persist(self, **kwargs):
         metadata, but now with the results fully computed or actively computing
         in the background.
 
+        The action of function differs significantly depending on the active
+        task scheduler.  If the task scheduler supports asynchronous computing,
+        such as is the case of the dask.distributed scheduler, then persist
+        will return *immediately* and the return value's task graph will
+        contain Dask Future objects.  However if the task scheduler only
+        supports blocking computation then the call to persist with *block*
```
Comment (jakirkham) [inline dask/base.py]: IIUC should be `will *block*`. PR ( https://github.com/dask/dask/pull/3305 ) should address this.
